oak_functions/wasm/src/lib.rs

```diff
@@ -292,19 +307,113 @@ where
     }
 }
 
+pub struct OakCaller<'a, L: OakLogger> {
+    caller: wasmi::Caller<'a, UserState<L>>,
+}
+
+pub struct Src {
+    ptr: AbiPointer,
+    len: AbiPointerOffset,
+}
+
+pub struct Dest {
+    ptr_ptr: AbiPointer,
+    len_ptr: AbiPointer,
+    buf: Vec<u8>,
+}
+
+impl<L> OakCaller<'_, L>
```

Comment:
I am not super convinced of the utility of this wrapper type (vs just a bunch of top-level functions that take `&mut Caller` as the first param. Also because some of them still do follow that signature (e.g. `read_buffer`). So if I were to add another function I am not sure whether it should be here or outside.

Up to you, perhaps let me know what was your main rationale for this wrapper.

Reply:
As mentioned below, I only have the "other" `read_buffer` for testing.

When I started to write `read_buffer_from_caller`, I thought of adding the wrapper type.

What I like about it: It is the dual to the `OakLinker` relying on `alloc` and `memory` as defined in Oak. If you are not against it, I'd keep it for now.